orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/pipeline/CanaryDisableClusterStage.groovy

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.kayenta.pipeline
+
+import com.netflix.spinnaker.orca.api.pipeline.CancellableStage
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageDefinitionBuilder
+import com.netflix.spinnaker.orca.api.pipeline.graph.StageGraphBuilder
+import com.netflix.spinnaker.orca.api.pipeline.graph.TaskNode
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution
+import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask
+import com.netflix.spinnaker.orca.mine.pipeline.CanaryDestroyClusterStage
+import com.netflix.spinnaker.orca.mine.pipeline.CanaryStage
+import com.netflix.spinnaker.orca.mine.tasks.DisableCanaryTask
+import com.netflix.spinnaker.orca.pipeline.WaitStage
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import javax.annotation.Nonnull
+
+import static com.netflix.spinnaker.orca.mine.pipeline.CanaryStage.DEFAULT_CLUSTER_DISABLE_WAIT_TIME
+
+@Component
+class CanaryDisableClusterStage implements StageDefinitionBuilder, CancellableStage {
+  public static final String STAGE_TYPE = "canaryDisableCluster"
+  @Autowired CanaryStage canaryStage
+  @Autowired WaitStage waitStage
+  @Autowired CanaryDestroyClusterStage canaryDestroyClusterStage
+
+  @Override
+  void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder builder) {
+    builder
+        .withTask("disableCanaryCluster", DisableCanaryTask)
+        .withTask("monitorDisable", MonitorKatoTask)
+        .withTask("disableBaselineCluster", DisableCanaryTask)
+        .withTask("monitorDisable", MonitorKatoTask)
+  }
+
+  @Override
+  void afterStages(StageExecution parent, StageGraphBuilder graph) {
+    Integer waitTime = parent.context.clusterDisableWaitTime != null ? stage.context.clusterDisableWaitTime : DEFAULT_CLUSTER_DISABLE_WAIT_TIME
+    graph.append {
```

Comment:
This is the real change here. We now use a wait stage instead of the wait task. The wait task was taking the stage's "startTime" as the basis for how long it should wait. Since the stage was the monitorCanary stage, if the canary ran longer than 180 seconds the wait task did nothing.